core/src/main/java/io/questdb/cairo/SymbolMapWriter.java

```diff
@@ -338,30 +340,26 @@ public void rebuildCapacity(
                 indexWriter.add(hash, offset);
             }
         } catch (Throwable th) {
-            // if we fail to rebuild files, we need to close them without truncates
-            // the files are links to previous version of symbol map
-            // truncating them will corrupt the symbol map we are trying to rebuild
-            if (charMem != null) {
-                charMem.close(false);
-            }
-            if (offsetMem != null) {
-                offsetMem.close(false);
-            }
-            if (indexWriter != null) {
-                indexWriter.closeNoTruncate();
-            }
+            closeNoTruncate();
             throw th;
         }
     }
 
     @Override
     public void rollback(int symbolCount) {
-        indexWriter.rollbackValues(keyToOffset(symbolCount - 1));
-        offsetMem.jumpTo(keyToOffset(symbolCount) + Long.BYTES);
-        jumpCharMemToSymbolCount(symbolCount);
-        valueCountCollector.collectValueCount(symbolIndexInTxWriter, symbolCount);
-        if (cache != null) {
-            cache.clear();
+        try {
+            indexWriter.rollbackValues(keyToOffset(symbolCount - 1));
+            offsetMem.jumpTo(keyToOffset(symbolCount) + Long.BYTES);
+            valueCountCollector.collectValueCount(symbolIndexInTxWriter, symbolCount);
+            if (cache != null) {
```

Comment:
nit: Misc.clear() performs null check

Reply:
done in [203603a](https://github.com/questdb/questdb/pull/5598/commits/203603aa7c248ccd9e4dcc322b7c31f8f642684d)